### ISSUE TITLE
Alerts - Link dashboard and panel with configurable dashboard parameters

### ISSUE THREAD

**Issue by Notedop** [grafana/grafana#94559]
**Why is this needed**:

When you have a dashboard that uses variables, then the default view of the dashboard may not immediately show the data that caused an alert. As the alert to dashboard linking does not allow for passing in dashboard variables, the user needs to take manual action to see the actual data that caused the alert.

**Issue example** 
Consider a dashboard that monitors windows OS, which has a dashboard variable "hostname". The dashboard only shows data for the selected hostname. Now imagine that you're monitoring 100 different Windows machines and you have a single generic rule on CPU usage. 

If it exceeds 80% for longer than 5 minutes trigger an alert. 

Now when the alert triggers, the user gets a mail, clicks on the dashboard or panel link and ends up on the dashboard or panel. But because the dashboard uses variables, the default dashboard variables are selected. So the initial view that the user sees, does not relate to the Hostname that caused the alert to trigger. The hostname was available as a label on the metric.

**What would you like to be added**:

Ability to map any label available during the alert (either labels configured on the Alert itself, or coming from the Metric that caused the alert) to a Dashboard variable with the goal that the dashboard URL and panel URL is dynamically generated when the alert is firing.

This applies to any feature that visualized the dashboard URL (email alerts, teams alerts, Alerts panel, etc).

Example:

Dashboard with dashboard variable Hostname exists. It contains the Hostname label values of metric "windows_cs_hostname". The dashboard has multiple panels which uses the hostname variable as filter.

When creating an alert and selecting the linked dashboard and panel should allow to optionally configure the dashboard variables for the link. The values of the dashboard variables can either be a static value or a value from a label of the metric that triggered the alert. For the latter this means that the URL is dynamically generated.

windows_cs_hostname{host="SRV123456"} = 95
windows_cs_hostname{host="SRV7891023"} = 95

Now if host SRV123456 has a high CPU usage that triggers the alert, the dashboard / panel link should be like this:
http://localhost:3000/d/1XaPI0Ink/windows-server-dashboard?var-hostname=SRV123456

Or if host SRV7891023 caused the alert then the dashboard / panel link should be 
http://localhost:3000/d/1XaPI0Ink/windows-server-dashboard?var-hostname=SRV7891023

There are multiple ways to achieve this for example using the [data link variables](https://grafana.com/docs/grafana/latest/panels-visualizations/configure-data-links/#data-link-variables) 

**Who is this feature for?**

Grafana administrators and Grafana end-users like support. Anyone that receives alerts.

**Comment by Notedop** [comment]
This is still relevant

**Comment by saibotk** [comment]
I want to add that this would also be extremely useful to make the image renderings actually usable for such alert notifications, when using the `capture = true` option for unified alerting.

**Comment by yuri-tceretian** [comment]
Indeed, this would be a good improvement. However, it will take a lot of efforts to change the approach the alerting handles relationship with dashboards. I put it in our backlog with lower priority.
